### ISSUE TITLE
MenuButton screen reader access issues

### ISSUE THREAD

**Issue by khmakoto** [microsoft/fluentui#21409]
The a11y review of the v9 `MenuButton` component (#20917) identified a bunch of screen reader access issues that need to be fixed:

- `aria-expanded` missing from `MenuButton` when collapsed.
